ecommerce/api.py

```diff
@@ -426,3 +428,48 @@ def pick_coupons(user):
             program_ids.add(program_id)
 
     return coupons
+
+
+def calculate_coupon_price(coupon, price, course_key):
+    """
+    Calculate the adjusted price given a coupon
+
+    Args:
+        coupon (Coupon): A coupon
+        price (decimal.Decimal): A price
+        course_key (str): An edX course key
+
+    Returns:
+        decimal.Decimal: An adjusted price
+    """
+    if course_key not in coupon.course_keys:
+        return price
+    if coupon.amount_type == Coupon.PERCENT_DISCOUNT:
+        return price * (1-coupon.amount)
+    elif coupon.amount_type == Coupon.FIXED_DISCOUNT:
+        return price - coupon.amount
+    else:
+        return price
+
+
+def calculate_run_price(course_run, user):
+    """
+    Calculate the price of a course given the coupons and financial aid available to the user.
+
+    Args:
+        course_run (CourseRun): A course run
+        user (django.contrib.auth.models.User): A user
+
+    Returns:
+        (decimal.Decimal, Coupon):
+            The adjusted of the course, and the coupon used if any
+    """
+    program = course_run.course.program
+    enrollment = get_object_or_404(ProgramEnrollment, program=program, user=user)
+    price = get_formatted_course_price(enrollment)['price']
+    coupons = [coupon for coupon in pick_coupons(user) if coupon.program == program]
+    coupon = None
```

Comment:
Nitpick: instead of setting `coupon = None` and then maybe populating the `coupon` variable, I would suggest returning early. Like this:

```python
coupons = [coupon for coupon in pick_coupons(user) if coupon.program == program]
if not coupons:
    return price, None
coupon = coupons[0]
price = calculate_coupon_price(coupon, price, course_run.edx_course_key)
return price, coupon
```